### PR TITLE
MVP-276: Component docs

### PR DESCRIPTION
**Pull request by BradyMitch**
## 🎯 Summary

  
[MVP-276](https://citz-imb.atlassian.net/jira/browse/MVP-276)

https://dev.developer.gov.bc.ca/docs/default/component/citz-imb-sso-css-api-npm-package/

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

File: techdocs/docs/using-the-package/apis-&-components/get-idir-users.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+## TypeScript Type
+
+```TypeScript
+(query: IDIRUserQuery) => Promise<GetUsersWithRoleResponse>;
+```
+
+The type `IDIRUserQuery`:
+
+* At least one property is required.
```
Comment (TaylorFries) [inline techdocs/docs/using-the-package/apis-&-components/get-idir-users.md]:
```suggestion
* At least one of the following properties are required.
```

File: techdocs/docs/using-the-package/apis-&-components/get-azure-idir-users.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+## TypeScript Type
+
+```TypeScript
+(query: IDIRUserQuery) => Promise<GetUsersWithRoleResponse>;
+```
+
+The type `IDIRUserQuery`:
+
+* At least one property is required.
```
Comment (TaylorFries) [inline techdocs/docs/using-the-package/apis-&-components/get-azure-idir-users.md]:
```suggestion
* At least one of the following properties are required.
```

File: techdocs/docs/using-the-package/apis-&-components/get-github-bcgov-users.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+## TypeScript Type
+
+```TypeScript
+(query: GitHubUserQuery) => Promise<GetUsersWithRoleResponse>;
+```
+
+The type `GitHubUserQuery`:
+
+* At least one property is required.
```
Comment (TaylorFries) [inline techdocs/docs/using-the-package/apis-&-components/get-github-bcgov-users.md]:
```suggestion
* At least one of the following properties are required.
```

File: techdocs/docs/using-the-package/apis-&-components/get-github-public-users.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+## TypeScript Type
+
+```TypeScript
+(query: GitHubUserQuery) => Promise<GetUsersWithRoleResponse>;
+```
+
+The type `GitHubUserQuery`:
+
+* At least one property is required.
```
Comment (TaylorFries) [inline techdocs/docs/using-the-package/apis-&-components/get-github-public-users.md]:
```suggestion
* At least one of the following properties are required.
```

File: mkdocs.yml [hunk 5]
```diff
@@ -11,6 +11,28 @@ nav:
...
+          - assignUserRoles: using-the-package/apis-&-components/assign-user-roles.md
+          - unassignUserRole: using-the-package/apis-&-components/unassign-user-role.md
+          - getUsersWithRole: using-the-package/apis-&-components/get-users-with-role.md
+          - getIDIRUsers: using-the-package/apis-&-components/get-idir-users.md
+          - getAzureIDIRUsers: using-the-package/apis-&-components/get-azure-idir-users.md
+          - getGitHubBCGovUsers: using-the-package/apis-&-components/get-github-bcgov-users.md
+          - getGitHubPublicUsers: using-the-package/apis-&-components/get-github-public-users.md
+          - getBasicBCeIDUser: using-the-package/apis-&-components/get-basic-bceid-user.md
+          - getBusinessBCeIDUser: using-the-package/apis-&-components/get-business-bceid-user.md
+          - getBothBCeIDUser: using-the-package/apis-&-components/get-both-bceid-user.md
```
Comment (TaylorFries) [inline mkdocs.yml]: should we consider breaking this into groups for better navigation?

**Review by TaylorFries — COMMENTED** [review]
For all the spots that have this can we consider using a note block for the "The Name column starting with * means the prop is required"
Screenshot 2024-06-05 at 8 44 53 AM

Also for these I made the note to change the phrasing on all of them but should we also make them note blocks?
Screenshot 2024-06-05 at 8 49 51 AM

File: mkdocs.yml (hunk 5 above)
Comment (BradyMitch) [reply mkdocs.yml]: Ya I'll give that a try!

**Comment by BradyMitch** [comment]
> For all the spots that have this can we consider using a note block for the "The Name column starting with * means the prop is required" Screenshot 2024-06-05 at 8 44 53 AM
> 
> Also for these I made the note to change the phrasing on all of them but should we also make them note blocks? Screenshot 2024-06-05 at 8 49 51 AM

Yes I think you're right, I'll try that out

**Review by TaylorFries — APPROVED** [review]
:fire: :fire: :fire:

**Pull request merged**